release notes/v0.5.0.md

```diff
@@ -0,0 +1,103 @@
+xk6-browser v0.5.0 is here! :tada:
+
+This minor release contains a new feature, a few important stability fixes, some breaking changes, improvements, and a continuation of our efforts to migrate to async APIs.
+
```

Comment:
We should mention the updated documentation and link it, possibly here so that it's prominent.

Reply:
With https://k6.io/docs/javascript-api/xk6-browser/? In that case we should merge and deploy the latest docs first.

Reply:
Yeah, but the link won't change, so we can mention it now and just make sure the docs deployment is done around the same time as our release.

Reply:
Resolved in: b5f262274131cc6b8c03496e705517aad39b8ba3